test/DOMINO/test_domino.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+import shutil
+from pathlib import Path
+from src.domino import DOMINO
+
+TEST_DIR = 'test/DOMINO/'
+OUT_FILE_DEFAULT = TEST_DIR+'output/domino-output.txt'
+OUT_FILE_OPTIONAL = TEST_DIR+'output/domino-output-thresholds.txt'
+OUT_FILE_PARSE = TEST_DIR+'output/domino-parse-output.txt'
+OUT_FILE_PARSE_EXP = TEST_DIR+'expected_output/domino-parse-output.txt'
+
+class TestDOMINO:
+    """
+    Run test for the DOMINO run function
+    """
+
+    def test_domino_required(self):
+        # Only include required arguments
+        out_path = Path(OUT_FILE_DEFAULT)
+        out_path.unlink(missing_ok=True)
+        DOMINO.run(
+            network=TEST_DIR+'input/domino-network.txt',
+            active_genes=TEST_DIR+'input/domino-active-genes.txt',
+            output_file=OUT_FILE_DEFAULT)
+        # output_file should be empty
+        assert out_path.exists()
+
+    def test_domino_optional(self):
+        # Include optional argument
+        out_path = Path(OUT_FILE_OPTIONAL)
+        out_path.unlink(missing_ok=True)
+        DOMINO.run(
+            network=TEST_DIR+'input/domino-network.txt',
+            active_genes=TEST_DIR+'input/domino-active-genes.txt',
+            output_file=OUT_FILE_OPTIONAL,
+            use_cache=False,
+            slices_threshold=0.4,
+            module_threshold=0.06)
+        # output_file should be empty
+        assert out_path.exists()
+
+    def test_domino_missing_active_genes(self):
+        # Test the expected error is raised when active_genes argument is missing
+        with pytest.raises(ValueError):
+            # No active_genes
+            DOMINO.run(
+                network=TEST_DIR+'input/domino-network.txt',
+                output_file=OUT_FILE_DEFAULT)
+
+    def test_domino_missing_network(self):
+        # Test the expected error is raised when network argument is missing
+        with pytest.raises(ValueError):
+            # No network
+            DOMINO.run(
+                active_genes=TEST_DIR+'input/domino-active-genes.txt',
+                output_file=OUT_FILE_DEFAULT)
+
+    def test_domino_parse_output(self):
+        # Show how the complicated file formats work in terms of input and expected output
+        # Concatenated module_0.html and module_1.html file as input
```

Comment:
Please add to the comment the dataset used to obtain these modules. Was it the EGFR dataset?

Reply:
Done, the dataset used is from https://github.com/Shamir-Lab/DOMINO/tree/master/examples.